HomeWork1/Alex_Zubok_home_work1.cpp

```diff
@@ -0,0 +1,60 @@
+#include <iostream>
+#include <limits>
+#include <cmath>
+
+int main() 
+{
+    long long var_a {};
+    size_t kMin = 1, kMax = 500;
+    std::cout << "Please enter value for variable 'a' from 1 till 500, and press 'Enter':";
```

Comment:
а в тексе тоже надо подправить) т.к. вы поменяете значение констант, но на экране вы увидете другие значения
```suggestion
    std::cout << "Please enter value for variable 'a' from " <<  kMin <<  " till " <<  kMax  << ", and press 'Enter': ";
```

Reply:
Так может тогда уже лучше использовать функцию?)

void check_valid_input( int &value, int Min, int Max ){
    while ( !( std::cin >> value ) || value < Min || value > Max ) {
        std::cin.clear();
        std::cin.ignore( std::numeric_limits<int>::max(), '\n' );
        std::cout << "Sorry, you enter invalid number of quantity inputs, please enter again[" << Min << " - " << Max << "]: ";
    }
}

Reply:
можете и функцию использовать. Вы когда этот код писали, мы еще функции не рассматривали. Вам решать).